### ISSUE TITLE
rethinkdb dump doesn't work on Windows

### ISSUE THREAD

**Issue by GlurG** [rethinkdb/rethinkdb#5505]
rethinkdb dump fails spectacularly on Windows with the following error:

```
Failed to create temporary directory (C:\Users\GlurG\AppData\Local\Temp\tmpois8c
kqo\rethinkdb_dump_2016-03-11T17:57:44_part): The filename, directory name, or v
olume label syntax is incorrect
Error: rethinkdb-export failed
```

I guess this happens because it tries to create a file with a ":" in it, which seems to be illegal on windows. I'm running the following version of RethinkDB

```
Running rethinkdb 2.2.4-windows-alpha-6 (debug) (MSC 190023506)...
Running on 6.1.7601 (Windows 7, Server 2008 R2)
```

**Comment by Wargog** [comment]
Right, there is a list of characters that aren't allowed when you create a folder in Explorer and `:` is one of them. Where did you find version alpha-6 though, you should probably try [alpha-4](https://download.rethinkdb.com/windows/rethinkdb-dev-preview-0.zip) which is the version listed on the website.

**Comment by danielmewes** [comment]
This is good to know, thanks for the feedback.

We haven't fully tested `rethinkdb dump` on Windows yet, but are trying to get it working for the 2.3 release.

The version of the server probably doesn't matter here, since the `rethinkdb dump` script actually comes from the Python driver.

**Comment by danielmewes** [comment]
@Tryneus Looks like we need to replace the `:` in the filename on Windows.

I suggest that we change this *only* on Windows (with a check in the code?), in case users on Linux or Mac are relying on the output filename in their setups.

**Comment by nighelles** [comment]
This is in next. OTS by @danielmewes

**Comment by GlurG** [comment]
Just tried this out again on RethinkDB 2.3.0 Windows Beta, and encountered the following error when executing "rethinkdb dump" using WinPython 3.4.3

```
C:\Projects\spoilstoolbox.com>rethinkdb dump

C:\Projects\spoilstoolbox.com>NOTE: 'rethinkdb-dump' saves data and secondary in
dexes, but does *not* save
 cluster metadata.  You will need to recreate your cluster setup yourself after

 you run 'rethinkdb-restore'.
Exporting to directory...
[========================================] 100%
6 rows exported from 4 tables, with 21 secondary indexes
  Done (8 seconds)
Zipping export directory...
Traceback (most recent call last):
  File "c:\program files\winpython-64bit-3.4.3.7\python-3.4.3.amd64\lib\runpy.py
", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\program files\winpython-64bit-3.4.3.7\python-3.4.3.amd64\lib\runpy.py
", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\WinPython-64bit-3.4.3.7\python-3.4.3.amd64\Scripts\reth
inkdb-dump.exe\__main__.py", line 9, in <module>
  File "c:\program files\winpython-64bit-3.4.3.7\python-3.4.3.amd64\lib\site-pac
kages\rethinkdb\_dump.py", line 200, in main
    run_rethinkdb_export(options)
  File "c:\program files\winpython-64bit-3.4.3.7\python-3.4.3.amd64\lib\site-pac
kages\rethinkdb\_dump.py", line 183, in run_rethinkdb_export
    do_zip(temp_dir, options)
  File "c:\program files\winpython-64bit-3.4.3.7\python-3.4.3.amd64\lib\site-pac
kages\rethinkdb\_dump.py", line 152, in do_zip
    is_fileobj = type(options["out_file"]) is file
NameError: name 'file' is not defined
```

However, "rethinkdb dump" works for Python 2.7.11 (on Windows)!

**Comment by danielmewes** [comment]
This seems to be the same issue as https://github.com/rethinkdb/rethinkdb/issues/5647 but for `rethinkdb dump`. Thanks for reporting it @GlurG .

**Comment by nltesown** [comment]
Has this been solved? I'm using RethinkDB 2.3.5 on Windows 7 and installed Python is 3.5.2. Running rethinkdb dump returns the message "failed to create process".
